src/psyclone/domain/gocean/nodes/gocean_extract_node.py

```diff
@@ -203,23 +208,25 @@ def create_driver(self, input_list, output_list):
         # name (e.g. dx_0).
 
         rename_variable = {}
-        for var_name in all_vars:
+        for signature in all_sigs:
             # TODO #644: we need to identify arrays!!
+
             # Support GOcean properties, which are accessed via a
             # derived type (e.g. 'fld%grid%dx'). In this stand-alone
             # driver we don't have the derived type, instead we create
             # variable based on the field in the derived type ('dx'
             # in the example above), and pass this variable to the
             # instrumented code.
-            last_percent = var_name.rfind("%")
-            if last_percent > -1:
+            var_name = str(signature)
+            if signature.is_structure:
```

Comment:
Funny that you say that. I originally had this all nice, no if-statement, and using `field_whole_dx` (etc) as local file names, which are more understandable names  (compared with just 'dx'), would allow us to support fields on different grids, ... but I ran into problems avoiding name clashes with a user variable `field_whole_dx`. I think once I create the driver in PSyIR, this should be do-able, but for now I reverted this issue ... a bit too much by also undoing the removal of the if statement.

Of course you are right, this if is not necessary.